tests/common.go

```diff
@@ -211,18 +211,18 @@ func ValidateContext(ctx *scheduler.Context, errChan ...*chan error) {
 			timeout = appScaleFactor * ctx.ReadinessTimeout
 		}
 
+		Step(fmt.Sprintf("wait for %s app to start running", ctx.App.Key), func() {
```

Comment:
actually, there was a specific reason why the WaitForRunning was after the validate volumes. 

If a PVC is in pending state, the application will never come up. While test will fail regardless, it will not be able to provide the exact reason of failure (i.e the volume creation was failing). 

Since your requirement is not to return if SkipVolumeValidation is true, you can change the step below to something like.

```
if !ctx.SkipVolumeValidation {
  ValidateVolumes(ctx, errChan...)
}
```

Reply:
@harsh-px done!